docs/design/redundant-suppressions.md

```diff
@@ -0,0 +1,94 @@
+
+This can be illustrated with the following example. Let us extend the above code to also print the value of `suffix` field.
+
+```csharp
+    [UnconditionalSuppressMessage("trim", "IL2070", Justification = "DynamicDependency attribute will instruct the linker to keep the public methods on NameProvider.")]
+    public void PrintName([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] Type type)
+    {
+        string name = (string)type.GetMethod("GetName")!.Invoke(null, null)!;
+        string suffix = (string)type.GetField("suffix")!.GetValue(null)!; // This is will issue a warning (only public methods are guaranteed to be kept).
+        Console.WriteLine(name);
+        Console.WriteLine(suffix);
+    }
+```
+
+Now the code is again trimmer-incompatible, the `GetField("suffix")` call will trigger a `IL2070` warning. However, as we forgot to remove the suppression silencing the `IL2070` warnings, the warning will be suppressed. We will not be informed about this trimmer-incompatible pattern during trimming.
+
+
+## Detecting redundant warning suppressions
+
+In order to avoid the above scenario, we would like to have an option to detect and report the warning suppressions which are not tied to any warnings caused by trim-incompatible patterns.
+
+This may be achieved by extending the linker tool functionality to check which suppression do in fact suppress warnings and reporting those which do not. 
+
+The said functionality can be implemented as an optional feature, triggered by passing a `
+--check-suppressions` command line argument to the linker. Running the tool with this option will report all of the warning suppressions which do not suppress any warnings.
+
+***NOTE:*** We will only process suppressions produced by the linker, other suppressions will be ignored.
+### Example:
+Let us again consider the example of the trimmer-compatible code with a redundant warning suppression. 
+
+```csharp
+    [UnconditionalSuppressMessage("trim", "IL2070", Justification = "DynamicDependency attribute will instruct the linker to keep the public methods on NameProvider.")] // This should be removed
+    public void PrintName([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] Type type)
+    {
+        string name = (string)type.GetMethod("GetName")!.Invoke(null, null)!;
+        Console.WriteLine(name);
+    }
+```
+
+In order to detect the warning suppression not tied to any trimmer-incompatible pattern we should run the `dotnet publish` command and pass the `--check-suppressions` option.
+```shell
+  dotnet publish -r win-x64 -p:PublishTrimmed=True -p:_ExtraTrimmerArgs="--check-suppressions"
+```
```

Comment:
I would probably remove this - just say, running the linker with the detection enabled will produce a new warning.

Reply:
done